runelite-client/src/main/java/net/runelite/client/plugins/inventorygrid/InventoryGridOverlay.java

```diff
@@ -72,10 +72,24 @@ private InventoryGridOverlay(InventoryGridConfig config, Client client, ItemMana
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
-		final Widget if1DraggingWidget = client.getIf1DraggedWidget();
-		final Widget inventoryWidget = client.getWidget(WidgetInfo.INVENTORY);
+		Widget draggingWidget = client.getIf1DraggedWidget();
+		final int draggedItemIndex;
+		final Widget inventoryWidget;
+		final boolean dynamic = draggingWidget == null;
 
-		if (if1DraggingWidget == null || if1DraggingWidget != inventoryWidget)
+		if (dynamic)
+		{
+			inventoryWidget = client.getWidget(WidgetInfo.BANK_INVENTORY_ITEMS_CONTAINER);
+			draggingWidget = client.getDraggedWidget();
+			draggedItemIndex = getDraggedIndex(inventoryWidget, draggingWidget);
+		}
+		else
+		{
+			inventoryWidget = client.getWidget(WidgetInfo.INVENTORY);
+			draggedItemIndex = client.getIf1DraggedItemIndex();
+		}
+
+		if (draggingWidget == null || inventoryWidget == null || draggingWidget.getId() != inventoryWidget.getId())
```

Comment:
Widgets are safe to reference compare, not sure why you have removed this here?

Reply:
It seems the draggedWidget and inventoryWidget instances are not the same when getting the widget from `client.getDraggedWidget()`

```
log.info("{} {} {} {} {} {}", dynamic, draggingWidget, inventoryWidget, draggingWidget.getId(), inventoryWidget.getId(), draggingWidget == inventoryWidget);

true hd@6e685864 hd@6e685864 9764864 9764864 true
false hd@12ba306b hd@50c028a2 983043 983043 false
```

Reply:
Added comment explaining why we're not comparing references.